lib/ecto/query/builder.ex

```diff
@@ -463,6 +489,8 @@ defmodule Ecto.Query.Builder do
   defp merge_fragments([h1], []),
     do: [{:raw, h1}]
 
+  @over_agg_1 ~w(row_number rank dense_rank percent_rank cume_dist)a
```

Comment:
Some of those return integers, don't they? for those, I think the correct return would be `{:any, :integer}`. We will also need to document those as aggregates in `Ecto.Query.API`.